packages/integrations/tailwind/README.md

```diff
@@ -127,7 +127,9 @@ export default {
 
 You can now [import your own `base.css` as a local stylesheet](https://docs.astro.build/en/guides/styling/#import-a-local-stylesheet).
 
-If you are using Vue, Svelte, or another component integration with Astro, `@apply` directives used in component `<style>`s may generate errors about your custom Tailwind class not existing and cause your builds to fail. [Instead of using `@layer` directives in a a global stylesheet](https://tailwindcss.com/docs/functions-and-directives#using-apply-with-per-component-css), define your custom styles by adding a plugin to your Tailwind config:
+When using the `@apply` directive in an Astro, Vue, Svelte, or another component integration's `<style>` tag, it may generate errors about your custom Tailwind class not existing and cause your builds to fail. Using nested classes like `dark:text-pink` would also not work in scoped `<style>`.
```

Comment:
Do we have a clearer idea about what exactly triggers these errors? I’ve used `@apply` in `.astro` successfully, e.g. <https://github.com/withastro/astro.build/blob/a4d8460976a36e335f2b0b4289e277b190750e19/src/components/CodeHero.astro#L65-L89>

Is it only nested classes that are an issue?

Reply:
If I understand correctly from previous reports, classes in `@apply` are also extracted into the main tailwind css file, so there's two copies of the class. Maybe it's not obvious in a large project where it's used in several places.

Reply:
Ahh, I see. Sounds like there are actually two (or three?) issues:

1. Some usage triggers “custom Tailwind class not existing” errors, breaking builds.

2. _Any_ usage can result in duplicate CSS, because Tailwind scans the `.astro` file to know which classes to include in the main bundle and is not smart enough to distinguish between a class name in frontmatter, a `class` attribute, etc. or inside `<style>`.

3. Nested classes like `dark:text-pink` are not compatible with `@apply` (is this actually the issue that causes 1?)

I think for **1** it would be amazing if we have more specific guidance than “may generate” if we know which usage is the issue (we may not of course).

For **2**, I think this can probably framed in terms of best practices like you’re doing in the second paragraph you added. Maybe we also state clearly there how you end up with duplicate CSS.

Reply:
Just tested no2 and it's false. Turns out tailwind strips of classes from `@apply` so not sure why someone mentioned it doesn't 😄 Will remove that.

For no1, it seems to be added by https://github.com/withastro/astro/pull/4353 about custom classes.

For no3, it's a separate issue from no1. `dark:text-pink` classes specifically is generated as `.dark > .text-pink`, but this doesn't match any element in the Astro markup so the compiler shakes them out.